File: solutions/go/rna-transcription/1/rna_transcription.go
package strand

var transcriber = map[rune]rune{
	'C': 'G',
	'G': 'C',
	'T': 'A',
	'A': 'U',
}

func ToRNA(dna string) string {
	var transcribed = make([]rune, len(dna))
	for i, n := range dna {
		transcribed[i] = transcriber[n]
	}
	return string(transcribed)
}

File: solutions/ocaml/rna-transcription/1/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let to_rna strand =
  List.map
    (
      fun x -> match x with
               | `G -> `C
               | `C -> `G
               | `T -> `A
               | `A -> `U
    )
    strand
